receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +90,489 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+				metricAttributes = columnOID.Attributes
+				metricResourceAttributes = columnOID.ResourceAttributes
+			}
+		}
+		indexString := strings.TrimPrefix(data.oid, data.parentOID)
+
+		// Set attributes for this metric's datapoint based on the previously gathered attributes.
+		// Keys will be determined from the related attribute config and values will come a few
+		// different places.
+		// Enum attribute value - comes from the metric config's attribute data
+		// Indexed prefix attribute value - comes from the current SNMP data's index and the attribute
+		// config's prefix value
+		// Indexed OID attribute value - comes from the previously collected indexed attribute data
+		// using the current index and attribute config to access the correct value
+		datapointAttributes := pcommon.NewMap()
+		for _, attribute := range metricAttributes {
+			attributeCfg := cfg.Attributes[attribute.Name]
+			attributeKey := attribute.Name
+			if attributeCfg.Value != "" {
+				attributeKey = attributeCfg.Value
+			}
+			attributeValue := attribute.Value
+			if attributeCfg.IndexedValuePrefix != "" {
+				attributeValue = attributeCfg.IndexedValuePrefix + indexString
+			} else if attributeCfg.OID != "" {
+				attrKey := indexedAttributeKey{
+					parentOID: attributeCfg.OID,
+					oidIndex:  indexString,
+				}
+				attributeValue = indexedAttributeValues[attrKey]
+				if attributeValue == "" {
+					return fmt.Errorf(errMsgOIDAttributeEmptyValue, metricName)
+				}
+			}
+			datapointAttributes.PutString(attributeKey, attributeValue)
+		}
+
+		resourceAttributes := map[string]string{}
+		resourceAttributeNames := []string{}
+		// Create a map of key/values for all related resource attributes. Keys will come directly from the
+		// metric config's resource attribute values. Values will come from the related attribute config's
+		// prefix value plus the index OR the previously collected resource attribute indexed data.
+		for _, attributeName := range metricResourceAttributes {
+			resourceAttributeNames = append(resourceAttributeNames, attributeName)
+			resourceAttributeCfg := cfg.ResourceAttributes[attributeName]
+
+			var attributeValue string
+			if resourceAttributeCfg.IndexedValuePrefix != "" {
+				attributeValue = resourceAttributeCfg.IndexedValuePrefix + indexString
+			} else if resourceAttributeCfg.OID != "" {
+				attrKey := indexedAttributeKey{
+					parentOID: resourceAttributeCfg.OID,
+					oidIndex:  indexString,
+				}
+				attributeValue = indexedResourceAttributeValues[attrKey]
+				if attributeValue == "" {
+					return fmt.Errorf(errMsgOIDResourceAttributeEmptyValue, metricName)
+				}
+			}
+			resourceAttributes[attributeName] = attributeValue
+		}
+
+		// Create a resource key using all of the relevant resource attribute names
+		sort.Strings(resourceAttributeNames)
+		resourceKey := ""
+		if len(resourceAttributeNames) > 0 {
+			resourceKey = strings.Join(resourceAttributeNames, "...") + indexString
+		}
```

Comment:
This doesn't seem right. A resource's attribute _names_ do not give it uniqueness. It needs to be based on _name/value pairs_.

Reply:
@djaglowski well it's the resource names/keys + the current SNMP data index for this metric value being looked at. so if there is a table with 10 rows, each row/index will correspond with a new resource for the same resource name.

Reply:
@djaglowski ok let me try to explain it a different way. lets say we have 3 resource attribute configs: `RA1`, `RA2`, `RA3` and 1 metric config `M1`. 
Lets say that `M1` has a single column OID in some table and has references to all 3 resource attribute configs. 
`RA1` and `RA2` also have an OID in their config which reference different columns in the same table. 
`RA3` has an indexed value prefix. 

So if that table has 3 rows, we are going to expect 3 different resources each with one 1 metric that contains 1 datapoint.

The three resources will have attributes that might look something like this:
Resource 1
RA1 => small ("small" is the value at index 1 for `RA1`'s OID column in the table)
RA2 => cold ("cold" is the value at index 1 for `RA2`'s OID column in the table)
RA3 => prefix.1 ("prefix.1" is created from `RA3`'s indexed prefix value + the index that corresponds with the related metric's datapoint
Metric M1 has 1 datapoint with value 10.0 (10.0 is the value at index 1 for `M1`'s column OID in the table)

Resource 2
RA1 => medium ("medium" is the value at index 2 for `RA1`'s OID column in the table)
RA2 => temperate ("temperate" is the value at index 2 for `RA2`'s OID column in the table)
RA3 => prefix.2 ("prefix.2" is created from `RA3`'s indexed prefix value + the index that corresponds with the related metric's datapoint
Metric M1 has 1 datapoint with value 15.0 (15.0 is the value at index 2 for `M1`'s column OID in the table)

Resource 3
RA1 => large ("large" is the value at index 3 for `RA1`'s OID column in the table)
RA2 => hot ("hot" is the value at index 3 for `RA2`'s OID column in the table)
RA3 => prefix.3 ("prefix.3" is created from `RA3`'s indexed prefix value + the index that corresponds with the related metric's datapoint
Metric M1 has 1 datapoint with value 5.0 (5.0 is the value at index 3 for `M1`'s column OID in the table)

So yes we could identify Resource 1 with a string key of "RA1=>small,RA2=>cold,RA3=>prefix.1" (which is what you are suggesting needs to be done).
But we also can uniquely identify Resource 1 with a "shortcut" string key of "RA1,RA2,RA3,1". Because the row index in the table is what is uniquely identifying a single resource within a single collection.

Reply:
Ok, thanks for explaining. To be honest, I think this is "trickier" than is should be, because we're managing OTel resources but using an abstract aspect of the snmp result to provide uniqueness. This is a good explanation though, so I think just adding it as a comment on the function is enough for now. Ideally this is converted to be fully OTel focused later so that any collector maintainer can more intuitively understand it.